std/typecons.d

```diff
@@ -5010,7 +5010,10 @@ assert(refCountedStore.isInitialized)).
         assert(rc1._refCounted._store._count == 2);
     }
     assert(p._refCounted._store == null);
+}
```

Comment:
This thing isn't technically `@safe`, because p is pointing at memory that is out of scope. You should leave this one as `@system`. What you can do is change the scope to be a `@safe` lambda. inside the `@system` unit test. Everything in here is safe except for this final check.